### ISSUE TITLE
FBAllocationTracker and related deps create libraries including the same source files

### ISSUE THREAD

**Issue by mikezucc** [pinterest/PodToBUILD#67]
FBMemoryProfiler, FBAllocationTracker, and FBRetainCycleDetector WORKSPACE rules create multiple objective c libraries that reference the same source. 

`FBAllocationTracker_cxx` collides with `FBAllocationTracker`

Pods.WORKSPACE: 
```python
new_pod_repository(
    name = "FBAllocationTracker",
    owner = "@vsolomenchuk",
    generate_module_map = False,
    enable_modules = False,
    url = "https://github.com/facebook/FBAllocationTracker/archive/0.1.3.zip",
)
```

Here is an example output BUILD file
```python
load(
  "//Vendor/rules_pods/BazelExtensions:extensions.bzl",
  "pch_with_name_hint",
  "acknowledged_target",
  "gen_module_map",
  "gen_includes"
  )
# Add a config setting release for compilation mode
# Assume that people are using `opt` for release mode
# see the bazel user manual for more information
# https://bazel.build/versions/master/docs/bazel-user-manual.html
native.config_setting(
  name = "release",
  values = {
    "compilation_mode": "opt"
  }
  )
filegroup(
  name = "FBAllocationTracker_cxx_hdrs",
  srcs = glob(
    [
      "FBAllocationTracker/**/*.h",
      "FBAllocationTracker/**/*.hpp",
      "FBAllocationTracker/**/*.hxx"
    ],
    exclude_directories = 1
    ),
  visibility = [
    "//visibility:public"
  ]
  )
filegroup(
  name = "FBAllocationTracker_cxx_union_hdrs",
  srcs = [
    "FBAllocationTracker_cxx_hdrs",
    "FBAllocationTracker_hdrs"
  ],
  visibility = [
    "//visibility:public"
  ]
  )
gen_includes(
  name = "FBAllocationTracker_cxx_includes",
  include = [
    "Vendor/FBAllocationTracker/pod_support/Headers/Public/"
  ]
  )
objc_library(
  name = "FBAllocationTracker_cxx",
  enable_modules = 0,
  srcs = glob(
    [
      "FBAllocationTracker/**/*.cc",
      "FBAllocationTracker/**/*.cpp",
      "FBAllocationTracker/**/*.cxx",
      "FBAllocationTracker/FBAllocationTracker.h",
      "FBAllocationTracker/FBAllocationTrackerFunctors.h",
      "FBAllocationTracker/FBAllocationTrackerImpl.h",
      "FBAllocationTracker/FBAllocationTrackerImpl.mm",
      "FBAllocationTracker/FBAllocationTrackerManager.h",
      "FBAllocationTracker/FBAllocationTrackerManager.mm",
      "FBAllocationTracker/FBAllocationTrackerSummary.h",
      "FBAllocationTracker/FBAllocationTrackerSummary.m",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.h",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.h",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.mm"
    ],
    exclude = [
      "FBAllocationTracker/**/*.S",
      "FBAllocationTracker/**/*.c",
      "FBAllocationTracker/**/*.m",
      "FBAllocationTracker/**/*.s"
    ],
    exclude_directories = 1
    ),
  non_arc_srcs = glob(
    [
      "FBAllocationTracker/**/*.mm"
    ],
    exclude = [
      "FBAllocationTracker/**/*.m",
      "FBAllocationTracker/FBAllocationTrackerImpl.mm",
      "FBAllocationTracker/FBAllocationTrackerManager.mm",
      "FBAllocationTracker/FBAllocationTrackerSummary.m",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.mm"
    ],
    exclude_directories = 1
    ),
  hdrs = [
    ":FBAllocationTracker_cxx_union_hdrs"
  ],
  pch = pch_with_name_hint(
    "FBAllocationTracker",
    [
      "FBAllocationTracker/**/*.pch"
    ]
    ),
  sdk_frameworks = [
    "Foundation"
  ],
  sdk_dylibs = [
    "c++"
  ],
  deps = [

  ] + [
    "FBAllocationTracker_cxx_includes"
  ],
  copts = [
    "-Wnon-modular-include-in-framework-module",
    "-g",
    "-stdlib=libc++",
    "-DCOCOAPODS=1",
    "-DOBJC_OLD_DISPATCH_PROTOTYPES=0",
    "-fdiagnostics-show-note-include-stack",
    "-fno-common",
    "-fembed-bitcode-marker",
    "-fmessage-length=0",
    "-fpascal-strings",
    "-fstrict-aliasing",
    "-Wno-error=nonportable-include-path"
  ] + select(
    {
      "//conditions:default": [
        "-DPOD_CONFIGURATION_RELEASE=0"
      ],
      ":release": [
        "-DPOD_CONFIGURATION_RELEASE=1",
        "-DNS_BLOCK_ASSERTIONS=1"
      ]
    }
    ) + [
    "-IVendor/FBAllocationTracker/pod_support/Headers/Public/FBAllocationTracker/"
  ] + [
    "-fmodule-name=FBAllocationTracker_pod_module"
  ],
  visibility = [
    "//visibility:public"
  ]
  )
acknowledged_target(
  name = "FBAllocationTracker_cxx_acknowledgement",
  deps = [

  ],
  value = "//Vendor/FBAllocationTracker/pod_support_buildable:acknowledgement_fragment"
  )
filegroup(
  name = "FBAllocationTracker_hdrs",
  srcs = glob(
    [
      "pod_support/Headers/Public/**/*"
    ],
    exclude_directories = 1
    ) + glob(
    [
      "FBAllocationTracker/**/*.h",
      "FBAllocationTracker/**/*.hpp",
      "FBAllocationTracker/**/*.hxx"
    ],
    exclude_directories = 1
    ) + [
    ":FBAllocationTracker_cxx_hdrs"
  ],
  visibility = [
    "//visibility:public"
  ]
  )
gen_includes(
  name = "FBAllocationTracker_includes",
  include = [
    "Vendor/FBAllocationTracker/pod_support/Headers/Public/"
  ]
  )
gen_module_map(
  "FBAllocationTracker",
  "FBAllocationTracker_module_map",
  "FBAllocationTracker",
  [
    "FBAllocationTracker_hdrs"
  ]
  )
objc_library(
  name = "FBAllocationTracker",
  enable_modules = 0,
  srcs = glob(
    [
      "FBAllocationTracker/**/*.S",
      "FBAllocationTracker/**/*.c",
      "FBAllocationTracker/**/*.s",
      "FBAllocationTracker/FBAllocationTracker.h",
      "FBAllocationTracker/FBAllocationTrackerFunctors.h",
      "FBAllocationTracker/FBAllocationTrackerImpl.h",
      "FBAllocationTracker/FBAllocationTrackerImpl.mm",
      "FBAllocationTracker/FBAllocationTrackerManager.h",
      "FBAllocationTracker/FBAllocationTrackerManager.mm",
      "FBAllocationTracker/FBAllocationTrackerSummary.h",
      "FBAllocationTracker/FBAllocationTrackerSummary.m",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.h",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.h",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.mm"
    ],
    exclude_directories = 1
    ),
  non_arc_srcs = glob(
    [
      "FBAllocationTracker/**/*.m"
    ],
    exclude = [
      "FBAllocationTracker/FBAllocationTrackerImpl.mm",
      "FBAllocationTracker/FBAllocationTrackerManager.mm",
      "FBAllocationTracker/FBAllocationTrackerSummary.m",
      "FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm",
      "FBAllocationTracker/Generations/FBAllocationTrackerGenerationManager.mm"
    ],
    exclude_directories = 1
    ),
  hdrs = [
    ":FBAllocationTracker_hdrs"
  ],
  pch = pch_with_name_hint(
    "FBAllocationTracker",
    [
      "FBAllocationTracker/**/*.pch"
    ]
    ),
  sdk_frameworks = [
    "Foundation"
  ],
  sdk_dylibs = [
    "c++"
  ],
  deps = [
    ":FBAllocationTracker_cxx"
  ] + [
    "FBAllocationTracker_includes"
  ],
  copts = [
    "-Wnon-modular-include-in-framework-module",
    "-g",
    "-stdlib=libc++",
    "-DCOCOAPODS=1",
    "-DOBJC_OLD_DISPATCH_PROTOTYPES=0",
    "-fdiagnostics-show-note-include-stack",
    "-fno-common",
    "-fembed-bitcode-marker",
    "-fmessage-length=0",
    "-fpascal-strings",
    "-fstrict-aliasing",
    "-Wno-error=nonportable-include-path"
  ] + select(
    {
      "//conditions:default": [
        "-DPOD_CONFIGURATION_RELEASE=0"
      ],
      ":release": [
        "-DPOD_CONFIGURATION_RELEASE=1",
        "-DNS_BLOCK_ASSERTIONS=1"
      ]
    }
    ) + [
    "-IVendor/FBAllocationTracker/pod_support/Headers/Public/FBAllocationTracker/"
  ] + [
    "-fmodule-name=FBAllocationTracker_pod_module"
  ],
  visibility = [
    "//visibility:public"
  ]
  )
acknowledged_target(
  name = "FBAllocationTracker_acknowledgement",
  deps = [

  ],
  value = "//Vendor/FBAllocationTracker/pod_support_buildable:acknowledgement_fragment"
  )
```

**Comment by rahul-malik** [comment]
@mikezucc - What is the actual error you're hitting? I just copied the allocations definition from your description and I'm seeing warnings that headers are in `srcs` but it still builds successfully

**Comment by jerrymarino** [comment]
Hey @rahul-malik @mikezucc thanks for filing this, I've finally got a fix up for the issue over in #139
